### ISSUE TITLE
RFE: Add Option to Skip Vault Variables in Inventory Import

### ISSUE THREAD

**Issue by jamesmarshall24** [ansible/awx#2597]
##### ISSUE TYPE
 - Feature Idea

##### COMPONENT NAME

 - API
 - UI

##### SUMMARY

Add check box to skip vaulted variables and variable files when importing inventory into AWX using an SCM inventory or similar

**Comment by wenottingham** [comment]
This is to just 'ignore them entirely and not error'?

**Comment by jamesmarshall24** [comment]
@wenottingham correct. Specifically for SCM inventory, there isn't an option to include vault credentials either and require a bit of a refactor from CLI vaulted files to include into AWX/Tower (without unencrypting into plain text).

**Comment by wenottingham** [comment]
Note that as we use `ansible-inventory` to do the inventory import, this may require changes in Ansible itself.

**Comment by AlanCoding** [comment]
> Specifically for SCM inventory, there isn't an option to include vault credentials

I think we need to create a new issue for this, aware of the current distinctions between in-line and whole-file encrypted variables. The current behavior to pass-through in-line variables is quite good from a usability perspective. They only time it _errors_ is for whole-file variables, and if we implement this feature, it would decrypt whole-file, and _only_ whole-file encrypted vars.

Otherwise, the model changes are very clear and we are well-prepared for it. The need switch from Popen to pexpect is fairly well understood, and I don't anticipate any major blockers for whole-file decryption on import. I don't know when we would do this, but I think it belongs on the agenda.

For this request, I would want to start with the Ansible core request. I wouldn't advocate for this feature myself, but if they do it, we will follow suit.

**Comment by wenottingham** [comment]
@AlanCoding that would be different, yes.

The goal of the initial issue is to have a checkbox such that whole-file vaulted content is ignored on inventory import. AFAIK right now it just throws an error.

**Comment by AlanCoding** [comment]
Yes. Does anyone mind if I put on the blocked label? As pointed out earlier, this could only come after core implementing it.

**Comment by thorian93** [comment]
As I just hit this *blocker* in my endeavour to get AWX up and running for our existing Ansible configuration: Any updates on this?

**Comment by azrdev** [comment]
related: 
* https://github.com/ansible/ansible/issues/58938
  * AIUI this would allow AWX to solve the issue, too
  * money quote:
    > The particular behavior of "don't decrypt vault values" can reasonably be accomplished with a simple flag on ansible-inventory

    (from related PR https://github.com/ansible/ansible/pull/37019#issuecomment-373811193 )
* https://github.com/ansible/ansible/issues/69858
* https://github.com/ansible/ansible/issues/7118

**Comment by AlanCoding** [comment]
I would call this a duplicate of https://github.com/ansible/awx/issues/12829, but the confusion lies in the degree of aggressiveness of the solution.

 - disable all vault variables (prevents pass-through-to-job behavior) - this issue
 - disable processing of vars plugins (prevents loading _unencrypted_ variables in those files)
 - ignore vault decryption errors

Only the last option is something I would consider to be a targeted solution. There's no reason to make the solution to this issue drop other unrelated variables.

**Comment by mateuszdrab** [comment]
Option `ignore vault decryption errors` would work from me. Encrypted vaults contain variables I do not want to be visible under hosts, they've been encrypted so that they're accessible at runtime but decrypting them and putting them under the host inventory item would defeat the point of having them encrypted.
I want to give some users ability to run some jobs without having them know those encrypted variables.
Any implementation of this would work better than having syncs fail for the past 6 months due to introduction of encrypted vaults in my env.

**Comment by anthonysr** [comment]
A workaround that a co-worker recommended, which worked was to only encrypt specific variable strings and not the entire YAML file. https://docs.ansible.com/ansible/latest/vault_guide/vault_encrypting_content.html.
